game/game.go

```diff
@@ -454,6 +465,32 @@ func (g *Game) GetPermanents(ids []PermanentId) []*Permanent {
 	return answer
 }
 
+// All objects must be put on the Stack via this method to get a unique ID.
+func (g *Game) AddToStack(stackObject *StackObject) {
+	stackObject.Id = g.NextStackObjectId
+	g.StackObjects[stackObject.Id] = stackObject
+	g.NextStackObjectId++
+	g.Stack = append(g.Stack, stackObject.Id)
+}
+
+func (g *Game) StackObject(id StackObjectId) *StackObject {
+	if id == StackObjectId(0) {
+		panic("0 is not a valid StackObjectId")
+	}
+	if g.StackObjects[id] == nil {
+		panic("id not found in g.StackObjects")
+	}
+	return g.StackObjects[id]
+}
+
+func (g *Game) GetStack(ids []StackObjectId) []*StackObject {
```

Comment:
does this function need an argument, or could the signature just be `func (g *Game) GetStack() []*StackObject` ?